app/src/main/java/de/westnordost/streetcomplete/MainActivity.java

```diff
@@ -416,6 +451,21 @@ private void confirmUndo(final OsmQuest quest)
 				if(isConnected()) downloadDisplayedArea();
 				else              Toast.makeText(this, R.string.offline, Toast.LENGTH_SHORT).show();
 				return true;
+			case R.id.action_open_location:
+				LatLon position = mapFragment.getPosition();
+				float zoom = mapFragment.getZoom();
+
+				Uri uri = Uri.parse(String.format("geo:%f,%f?z=%f",
+					position.getLatitude(),
+					position.getLongitude(),
+					zoom
+				));
+
+				intent = new Intent(Intent.ACTION_VIEW, uri);
+				if (intent.resolveActivity(getPackageManager()) != null) {
+					startActivity(intent);
```

Comment:
What actually happens here? Will an activity selector be shown, or will the app previously selected as "primary geolocation app" be started?

Also, if there is no activity that takes a geo intent, and the Android system does not already display the lack of it in some way (showing an empty app selection screen?), perhaps a toast message should be shown instead.

Reply:
> Will an activity selector be shown, or will the app previously selected as "primary geolocation app" be started?

Both. If the user selected a default app to handle a Geo URI before, this app opens without the activity selector. Otherwise you can select an application from the list.

> Also, if there is no activity that takes a geo intent

Right now, nothing will happen at all... Is it better to display a toast message or remove the item from the options menu if the intent can't be handled by any other activity?

Reply:
I think it is better to display a short toast message in that case than hide the option, e.g. "No map application installed" or similar.

Reply:
Maybe "No _other_ map application installed" (would be self-contradictory otherwise, as SC is such a mapping app).

Reply:
No other map application installed is now the text of the toast

Reply:
Wait, right, so if StreetComplete can interpret a geo URI, then the case that no application exists that will take the intent does not exist, or is the source activity automatically excluded there?

Reply:
It is excluded. But if you have both versions installed (debug and release) then you can of course switch between both... :smile: